### ISSUE TITLE
Window dragging too slow due to re-rendering of inner content

### ISSUE THREAD

**Issue by mgrouch** [rjaros/kvision#367]
Window dragging too slow due to re-rendering of inner content.

Can inner content be removed while dragging//resizing and masked?
When window is resized or dragging stopped inner part needs to be rendered back automatically.

Thanks

**Comment by rjaros** [comment]
You can easily hide the content while dragging with two event handlers.
```Kotlin
class MyWindow : Window("Title", "fas fa-paint-brush", 700, 400) {
    init {
        val innerContent = simplePanel {
        // the window content in here
        }
        onEvent {
            dragStartWindow = {
                innerContent.hide()
            }
            dragEndWindow = {
                innerContent.show()
            }
        }
    }
}
```

**Comment by mgrouch** [comment]
Thanks. It works

**Comment by mgrouch** [comment]
Actually it doesn’t work on resize. 
I can’t find resize start and resize end event handlers. 

Thanks

**Comment by rjaros** [comment]
Resize is implemented with standard CSS resize property and unfortunately there are no standard events to detect begin and end of the resize process. There are some workaround for this problem: https://stackoverflow.com/questions/5489946/how-to-wait-for-the-end-of-resize-event-and-only-then-perform-an-action, I will play a bit with this issue and let you know.

**Comment by mgrouch** [comment]
Another issue is that drag event is triggered when I just click in window title. I haven’t dragged anything. So the code above makes windows flicker when I just click around on windows titles. Kind of not working for my targeted user base. 

Another issue with desktop example with zIndex manipulation. toFront() which adds +1 and other methods which add 1000 just don’t mix. toFront() should be used everywhere instead. 

One more issue. Resize event is triggered too on a window behind a window I click to minimize. 

I’ve also seen constant resize events triggered and window flicker when I resize and do not let go of left mouse button pressed. 
the window would go resizing back and forth few pixels.

**Comment by rjaros** [comment]
> Another issue is that drag event is triggered when I just click in window title. I haven’t dragged anything. So the code above 
> makes windows flicker when I just click around on windows titles. Kind of not working for my targeted user base.

This will be fixed in the next release.

> Another issue with desktop example with zIndex manipulation. toFront() which adds +1 and other methods which add 
> 1000 just don’t mix. toFront() should be used everywhere instead.

The minimize/maximize functionality is not implemented in the framework itself. There are just empty methods to be overridden or dispatched events to be handled. The simple implementation with adding big number to the `zIndex` is just for the demo application (just to be sure the window is over appbar component, which is at `zIndex` 1030 from Bootstrap CSS). If you will need to implement this yourself, you are free to do this any way you like.

> One more issue. Resize event is triggered too on a window behind a window I click to minimize.
> I’ve also seen constant resize events triggered and window flicker when I resize and do not let go of left mouse 
> button pressed. the window would go resizing back and forth few pixels.

I've re-written the `Window` component with the modern [Resize Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API) instead of the old, external JS library. I've also implemented `resizeStartWindow` event. So you can now use:
```
onEvent {
            dragStartWindow = {
                innerContent.hide()
            }
            dragEndWindow = {
                innerContent.show()
            }
            resizeStartWindow = {
                innerContent.hide()
            }
            resizeWindow = {
                innerContent.show()
            }
        }
```
This seems to fix the flickering effect visible with some browsers as well (at least for me).

All these changes will be available in 5.9.1.

**Comment by mgrouch** [comment]
Thanks.

**Comment by rjaros** [comment]
Released with 5.10.0
